### PR TITLE
Import histories

### PR DESCRIPTION
**Pull request by davidchristiany**
wip: method to import history from an archive or an url.

### CODE REVIEW

File: bioblend/galaxy/histories/__init__.py [hunk 1]
```diff
@@ -37,6 +38,16 @@ def create_history(self, name=None):
             payload['name'] = name
         return self._post(payload)
 
+    def import_history(self, file_path=None, url=None, files_attached=False):
+        if file_path:
+            files_attached=True
+            archive_file=attach_file(file_path)
+            payload = dict(archive_source='', archive_file=archive_file)
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion
            payload = dict(archive_source='', archive_file=archive_file, archive_type="file")
```
The archive_type is passed on to the `imp_history_from_archive` tool

File: bioblend/galaxy/histories/__init__.py [hunk 2]
```diff
@@ -37,6 +38,16 @@ def create_history(self, name=None):
             payload['name'] = name
         return self._post(payload)
 
+    def import_history(self, file_path=None, url=None, files_attached=False):
+        if file_path:
+            files_attached=True
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion
            files_attached = True
```

File: bioblend/galaxy/histories/__init__.py [hunk 3]
```diff
@@ -37,6 +38,16 @@ def create_history(self, name=None):
...
+            archive_file=attach_file(file_path)
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion
            archive_file = attach_file(file_path)
```

File: bioblend/galaxy/histories/__init__.py [hunk 4]
```diff
@@ -37,6 +38,16 @@ def create_history(self, name=None):
...
+            payload = dict(archive_source='', archive_file=archive_file)
+        else:
+            payload = dict(archive_source=url, archive_type='url')
+
+        return self._post(payload=payload,files_attached=files_attached)
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion
        return self._post(payload=payload, files_attached=files_attached)
```

**Comment by davidchristiany** [comment]
I tried with  `archive_type="file"` in the payload but still got the same error:
` Error unpacking tar/gz archive: [Errno 2] No such file or directory: '/Users/pret/galaxy/database/tmp/tmpw3cvn5av'`

File: bioblend/galaxy/histories/__init__.py [hunk 5]
```diff
@@ -37,6 +38,16 @@ def create_history(self, name=None):
             payload['name'] = name
         return self._post(payload)
 
+    def import_history(self, file_path=None, url=None, files_attached=False):
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]: Can you add a docstring here, something like
```suggestion
    def import_history(self, file_path=None, url=None):
        """
        Import a history from an archive on disk or a URL.
        
        :type file_path: str
        :param file_path: Path to exported history archive on disk.
        
        :type url: str
        :param url: URL for an exported history archive
        """
```
And drop the `files_attached` parameter, you can set this in the function body

File: bioblend/galaxy/histories/__init__.py [hunk 6]
```diff
@@ -37,6 +38,16 @@ def create_history(self, name=None):
...
+        if file_path:
+            files_attached = True
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion
```

File: bioblend/galaxy/histories/__init__.py [hunk 7]
```diff
@@ -37,6 +38,16 @@ def create_history(self, name=None):
...
+            archive_file = attach_file(file_path)
+            payload = dict(archive_source='', archive_file=archive_file, archive_type="file")
+        else:
+            payload = dict(archive_source=url, archive_type='url')
+
+        return self._post(payload=payload, files_attached=files_attached)
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion
        return self._post(payload=payload, files_attached=file_path is not None)
```

**Comment by davidchristiany** [comment]
I' will make a testcase, in bioblend/tests/TestGalaxyHistories.py I presume ?

**Comment by mvdbeek** [comment]
yes

File: bioblend/galaxy/histories/__init__.py [hunk 8]
```diff
@@ -37,6 +38,24 @@ def create_history(self, name=None):
             payload['name'] = name
         return self._post(payload)
 
+    def import_history(self, file_path=None, url=None):
+        """
+        Import a history from an archive on disk or a URL.
+
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion

```

File: bioblend/galaxy/histories/__init__.py [hunk 9]
```diff
@@ -37,6 +38,24 @@ def create_history(self, name=None):
             payload['name'] = name
         return self._post(payload)
 
+    def import_history(self, file_path=None, url=None):
+        """
+        Import a history from an archive on disk or a URL.
+        
+        :type file_path: str
+        :param file_path: Path to exported history archive on disk.
+
```
Comment (mvdbeek) [inline bioblend/galaxy/histories/__init__.py]:
```suggestion

```

**Comment by nsoranzo** [comment]
The failing jobs for releases 16.04-17.05 will be fixed by https://github.com/galaxyproject/galaxy/pull/9089 .

File: bioblend/_tests/TestGalaxyHistories.py [hunk 10]
```diff
@@ -183,6 +183,9 @@ def test_download_history(self):
         finally:
             shutil.rmtree(tempdir)
 
+    def test_import_history(self):
+        self.gi.histories.import_history(file_path='tests/data/Galaxy-History-test.tar.gz')
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyHistories.py]:
```suggestion
        path = test_util.get_abspath(os.path.join('data', 'Galaxy-History-test.tar.gz'))
        self.gi.histories.import_history(file_path=path)
```

Comment (nsoranzo) [reply bioblend/_tests/TestGalaxyHistories.py]: Ping @davidchristiany

**Pull request merged**